application/src/test/java/today/stepbeyond/examples/springbootexamples/application/domain/BirthOfDogIT.java

```diff
@@ -0,0 +1,101 @@
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.cloud.contract.wiremock.AutoConfigureWireMock;
+import org.springframework.jms.core.JmsTemplate;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.utility.DockerImageName;
+import today.stepbeyond.examples.springbootexamples.domain.model.Cat;
+import today.stepbeyond.examples.springbootexamples.domain.model.Dog;
+import today.stepbeyond.examples.springbootexamples.domain.model.Pet;
+import today.stepbeyond.examples.springbootexamples.infrastructure.gateways.events.PetEventPublisher;
+import today.stepbeyond.examples.springbootexamples.infrastructure.gateways.events.model.PetEvent;
+import today.stepbeyond.examples.springbootexamples.infrastructure.gateways.events.model.PetIsBornEvent;
+import today.stepbeyond.examples.springbootexamples.infrastructure.gateways.events.model.PetRegisteredEvent;
+import today.stepbeyond.examples.springbootexamples.infrastructure.repository.PetRepository;
+import today.stepbeyond.examples.springbootexamples.usecases.BirthOfADog;
+import wiremock.org.eclipse.jetty.http.HttpHeader;
+
+@SpringBootTest(
+    properties = {"petStoreApi.url=http://localhost:${wiremock.server.port}/"},
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureWireMock(port = 0)
+class BirthOfDogIT {
+
+  @Container
+  public GenericContainer<?> activemq =
+      new GenericContainer(DockerImageName.parse("rmohr/activemq:5.14.3")).withExposedPorts(61616);
+
+  @Autowired PetRepository petRepository;
+
+  // With @SpyBean, you can check your beans with mockito toolset
+  @SpyBean PetEventPublisher petEventPublisher;
+
+  @Autowired BirthOfADog testee;
+
+  @Autowired JmsTemplate template;
+
+  @Test
+  void checkIfDogIsBorn() {
+
+    // GIVEN
+    WireMock.stubFor(
+        post("/registrations")
+            .willReturn(
+                created()
+                    .withHeader(HttpHeader.LOCATION.asString(), UUID.randomUUID().toString())));
+
+    // WHEN
+    var newDog = testee.birthOfADog("Lilly");
+
+    // THEN
+
+    // With soft assertions you can disable the default "fail fast" of the asstion framework
+    assertSoftly(
+        softly -> {
+          softly
+              .assertThat(petRepository.find(newDog.getId()))
+              .isPresent()
+              .get()
+              .isExactlyInstanceOf(Dog.class)
+              .extracting(Pet::getName)
+              .isEqualTo("Lilly");
+          softly
+              .assertThat(petRepository.find(newDog.getId()))
+              .isPresent()
+              .get()
+              .isNotInstanceOf(Cat.class);
```

Comment:
NIT: I think there is something like `containsExactlyInstanceOf(Cat.class)` in assertj library